Apps/Src/Tasks.c

```diff
@@ -60,6 +61,65 @@ OS_SEM Fault_Sem4;
 cell_asic Minions[NUM_MINIONS];
 uint32_t WDog_BitMap = 0;
 
+
+/** Pend_ResourceForCLI
+ * Pends for desired resource that CLI needs
+ * @param resourceNum the resource you want to wait for, look at CLI_Resource enum in CLI.h
+ */
+void Pend_ResourceForCLI(int resourceNum){
+    switch(resourceNum){
+        case 0:
+            OS_Sem4Pend(&SafetyCheck_Sem4);
+            break;
+        case 1:
+            OS_MutexLock(&MinionsASIC_Mutex);
+            break;
+        case 2:
+            OS_MutexLock(&VoltageBuffer_Mutex);
+            break;
+        case 3:
+            OS_MutexLock(&Temperature_Mutex);
+            break;
+        case 4:
+            OS_MutexLock(&OpenWireBuffer_Mutex);
+            break;
+        case 5:
+            OS_MutexLock(&AmperesData_Mutex);
+            break;
+        default:
+            break;
+    }
+}
+
+/** Post_ResourceForCLI
+ * Posts desired resource that CLI needs
+ * @param resourceNum the resource you want to post, look at CLI_Resource enum in CLI.h
+ */
+void Post_ResourceForCLI(int resourceNum){
```

Comment:
Should this also be in CLI.c?

Reply:
I made the Pend_ResourceForCLI() and Post_ResourceForCLI() functions because I thought all the semaphores/mutexes were going to be global variables declared in Tasks.c. I put these functions in Tasks.h so that way I could access the Task module's globals in the CLI module.

Reply:
I think we could do this 2 ways. We could define the semaphores and mutexes in tasks.h and keep our functions as static in our specific c files (becuase if every thread had their own pend and post function in tasks.c it might get cluttered). Another idea could be to have these functions as just Post_Resource and Pend_Resource and any thread could call them to check these semaphores. What do you and @ClarkPoon think?

Reply:
I think we should do the latter.

Reply:
Regardless of if we do the second or the first one, I would recommend that the variable passed into these functions be the type of the enum that you are passing into the functions (rather than an int). I would also make each case in the switch case statement one of the enum types rather than an integer, just so it is easier to read.

Reply:
We should be only pending and posting the mutexes in the functions that directly access the shared resource. In this case, you are calling a different function that does the actual access, so you should not pend or post the mutexes here (as we discussed at the meeting yesterday).